api/graphql/tests/test_applications/test_application_create.py

```diff
@@ -0,0 +1,255 @@
+    def test_application_create_organization_address_cannot_be_empty(self):
+        self.client.force_login(self.regular_joe)
+        data = self.get_application_data()
+        data["organisation"].update({"address": dict()})
+        assert_that(Application.objects.count()).is_equal_to(1)
+
+        response = self.query(self.get_create_query(), input_data=data)
+        assert_that(response.status_code).is_equal_to(400)
+
+        assert_that(Application.objects.count()).is_equal_to(1)
+
+    def test_application_create_organization_address_not_included(self):
+        self.client.force_login(self.regular_joe)
+        data = self.get_application_data()
+        data["organisation"].pop("address")
+        assert_that(Application.objects.count()).is_equal_to(1)
+
+        response = self.query(self.get_create_query(), input_data=data)
+        assert_that(response.status_code).is_equal_to(400)
+
+        assert_that(Application.objects.count()).is_equal_to(1)
+
+    def test_application_create_organization_not_included(self):
+        self.client.force_login(self.regular_joe)
+        data = self.get_application_data()
+        data["organisation"] = None
+        assert_that(Application.objects.count()).is_equal_to(1)
+
+        response = self.query(self.get_create_query(), input_data=data)
+        assert_that(response.status_code).is_equal_to(200)
+
+        content = json.loads(response.content)
+        app_data = content.get("data").get("createApplication")
+        assert_that(app_data.get("errors")).is_none()
+        assert_that(Application.objects.count()).is_equal_to(2)
+        application = Application.objects.get(id=app_data.get("pk"))
+        assert_that(application.organisation).is_none()
+
+    def test_application_create_organization_email_can_be_empty(self):
+        self.client.force_login(self.regular_joe)
+        data = self.get_application_data()
+        data["organisation"].update({"email": ""})
+
+        response = self.query(self.get_create_query(), input_data=data)
+        assert_that(response.status_code).is_equal_to(200)
+
+        content = json.loads(response.content)
+        app_data = content.get("data").get("createApplication")
+        assert_that(app_data.get("errors")).is_none()
+        assert_that(Application.objects.count()).is_equal_to(2)
+        application = Application.objects.get(id=app_data.get("pk"))
+        assert_that(application.organisation.email).is_empty()
+
+    def test_application_create_organization_email_cannot_be_null(self):
+        self.client.force_login(self.regular_joe)
+        data = self.get_application_data()
+        data["organisation"].update({"email": None})
+
+        response = self.query(self.get_create_query(), input_data=data)
+        assert_that(response.status_code).is_equal_to(200)
+
+        content = json.loads(response.content)
+        app_data = content.get("data").get("createApplication")
+        assert_that(app_data.get("errors")).is_not_none()
```

Comment:
Most failure tests check that the data was not updated to the database. Perhaps it could be useful here as well:
```
assert_that(Application.objects.count()).is_equal_to(1)
```

Reply:
added this